dataset/include/scipp/dataset/bucket.h

```diff
@@ -22,6 +23,25 @@ SCIPP_DATASET_EXPORT void copy_slices(const DatasetConstView &src,
 [[nodiscard]] SCIPP_DATASET_EXPORT Dataset resize_default_init(
     const DatasetConstView &parent, const Dim dim, const scipp::index size);
 
+[[nodiscard]] SCIPP_DATASET_EXPORT Variable from_constituents(Variable indices,
```

Comment:
Newly added functions are quite self-explanatory IMHO, but `resize_default_init` could use some comments to describe what it's doing and what the parameters are for.

Reply:
Added to doxygen in `.cpp`. Note that the function will likely get renamed, I wrote the doxygen following the future rename buckets->bins.